bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/firmware/Firmware.java

```diff
@@ -231,7 +233,7 @@ public String getMd5Hash() {
                 try (DigestInputStream dis = new DigestInputStream(inputStream, md)) {
                     bytes = IOUtils.toByteArray(dis);
                 } catch (IOException ioEx) {
-                    LOGGER.error("Cannot read firmware with UID {}.", uid, ioEx);
+                    LOGGER.error("Cannot read firmware {}.", this, ioEx);
```

Comment:
Btw, why is this logger static? Could you please make it non-static & lower-case?

Reply:
Originally I wanted to avoid having a new logger created for each new instance of this class - especially due to potential performance issues on embedded devices.

Reply:
See also:
* https://github.com/jupnp/jupnp/pull/68#pullrequestreview-3429591
* https://stackoverflow.com/a/10456613

Reply:
I will fix it as @sjka suggested, thak you for the participation into the discussion!